dhcpv4/ztpv4/ztp.go

```diff
@@ -15,39 +17,59 @@ type VendorData struct {
 
 var errVendorOptionMalformed = errors.New("malformed vendor option")
 
-// ParseVendorData will try to parse dhcp4 options looking for more
-// specific vendor data (like model, serial number, etc).
-func ParseVendorData(packet *dhcpv4.DHCPv4) (*VendorData, error) {
+func parseVIVC(vd *VendorData, packet *dhcpv4.DHCPv4) error {
+	for _, id := range packet.VIVC() {
+		if id.EntID == uint32(iana.EntIDCiscoSystems) {
+			vd.VendorName = iana.EntIDCiscoSystems.String()
+			//SN:0;PID:R-IOSXRV9000-CC
+			for _, f := range bytes.Split(id.Data, []byte(";")) {
+				p := bytes.Split(f, []byte(":"))
+				if len(p) != 2 {
+					return errVendorOptionMalformed
+				}
+
+				switch string(p[0]) {
+				case "SN":
+					vd.Serial = string(p[1])
+				case "PID":
+					vd.Model = string(p[1])
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func parseVendorClass(vd *VendorData, packet *dhcpv4.DHCPv4) error {
```

Comment:
I am not a fan of modifying the parameter, can it return a `*VendorData` instead and make it consistent with the `ParseVendorData` function definition?